lib/fs.js

```diff
@@ -2070,11 +2074,43 @@ function writeAll(fd, isUserFd, buffer, offset, length, signal, callback) {
     } else {
       offset += written;
       length -= written;
-      writeAll(fd, isUserFd, buffer, offset, length, signal, callback);
+      writeAll(
+        fd, isUserFd, buffer, offset, length, signal, encoding, callback);
     }
   });
 }
 
+async function writeAllCustomIterable(
+  fd, isUserFd, buffer, offset, length, signal, encoding, callback) {
+  const result = await buffer.next();
+  if (result.done) {
+    fs.close(fd, callback);
+    return;
+  }
+  const resultValue = result.value.toString();
```

Comment:
You should convert buffers to string, and not the other way around. See how it's done in the beginning of `writeFile`

Reply:
also, it probably makes sense to check if the signal was aborted here.